solutions/swb-reference/src/services/dataSetService.ts

```diff
@@ -319,6 +331,16 @@ export class DataSetService implements DataSetPlugin {
 
     const response = await this.addAccessPermission(permissionRequest);
 
+    if (request.accessLevel === 'read-write') {
+      await this._addAuthZPermissionsForDataset(
+        request.authenticatedUser,
+        SwbAuthZSubject.SWB_DATASET_UPLOAD,
+        request.dataSetId,
+        [projectAdmin, projectResearcher],
+        ['READ']
```

Comment:
should it be ['READ', 'WRITE'] here?

Reply:
Nope, the endpoint uses a `GET`, so it only needs to be a read.